pydexarm/pydexarm.py

```diff
@@ -2,128 +2,258 @@
 import re
 
 class Dexarm:
+    """ Python class for Dexarm
+    """
 
     def __init__(self, port):
+        """
+        Args:
+            port (string): the serial port of Dexarm, e.g, "COM3"
+        """
         self.ser = serial.Serial(port, 115200, timeout=None)
         self.is_open = self.ser.isOpen()
         if self.is_open:
             print('pydexarm: %s open' % self.ser.name)
         else:
-            print('Failed to open serial port')
+            print('failed to open serial port')
 
-    def _send_cmd(self, data):
+    def _send_cmd(self, data, wait=True):
+        """
+        Send command to the arm.
+
+        Args:
+            data (string): the command
+            wait (bool): wait for response from the arm (ok) or not.
+                If True, this function will block until the arm response "ok"
+                If False, this function will not block here. But the command could be ignored if buffer of the arm is full.
+        """
+        self.ser.reset_input_buffer()
         self.ser.write(data.encode())
+        if not wait:
+            return
         while True:
-            str = self.ser.readline().decode("utf-8")
-            if len(str) > 0:
-                if str.find("ok") > -1:
+            serial_str = self.ser.readline().decode("utf-8")
+            if len(serial_str) > 0:
+                if serial_str.find("ok") > -1:
                     print("read ok")
                     break
                 else:
-                    print("read：", str)
+                    print("read：", serial_str)
 
     def go_home(self):
+        """
+        Go to home position and enable the motors. Should be called each time when power on.
+        """
         self._send_cmd("M1112\r")
 
     def set_workorigin(self):
+        """
+        Set the work origin.
```

Comment:
I'm not sure about the effect of this function. please give a detailed description.